### ISSUE TITLE
lint: test helper function gives lint warning if context is not first parameter

### ISSUE THREAD

**Issue by posener** [golang/lint#422]
A test helper function for a context object causes "warning: context.Context should be the first parameter of a function".

```go
func testTheContext(t *testing.T, ctx context.Context) {
    t.Helper()
    // test ctx...
}
```

I argue that in this case, the context shouldn't be the first argument.

**Comment by katiehockman** [comment]
I spoke with @andybons, and looking at the [scope for lint contributions](https://github.com/golang/lint#scope), the only changes being accepted at this point are those that help meet the goal of carrying out the stylistic conventions put forth in Effective Go and CodeReviewComments.

Potentially this is a change that can be made to those documents, but as it is now, this issue should be closed.

**Comment by posener** [comment]
Hi,
I don't see why this is not aligned with the scope of contributions - the current behavior is wrong.
How can I appeal on this decision?

**Comment by andybons** [comment]
@posener can you show us specific examples from either [Effective Go](https://golang.org/doc/effective_go.html) or [CodeReviewComments](https://golang.org/wiki/CodeReviewComments) that support your proposed behavior?

**Comment by posener** [comment]
Sorry, I now understand what you meant.
Does this means that the right way for this function signature is `func testTheContext(ctx context.Context, t *testing.T)`? It doesn't look right to me.
Should CodeReviewComments instruct that test helper functions should always have their first argument as `t *testing.T`?

**Comment by katiehockman** [comment]
As the documentation is now, yes `func testTheContext(ctx context.Context, t *testing.T)` would be right. If you want to propose a change to CodeReviewComments, then that would be more appropriate in a different issue.

**Comment by posener** [comment]
Sure, where should this issue be opened?

**Comment by katiehockman** [comment]
This would be a change to the wiki, which would be an issue filed in the golang/go repository. Something similar to https://github.com/golang/go/issues/29289.
